### PR TITLE
The Toxins Update - New machine & miner map tweaks

### PR DESCRIPTION
**Pull request by MonsWTF**
### Intent of your Pull Request

This PR adds a new machine to Toxins that resolves the issues plaguing interactions between Toxins Science and Mining. 

Introducing L.A.M., the Lavaland Artillery Mainframe. (spritework by Tipy)

LAM

The L.A.M. provides Toxins scientists a new, direct way to aid Miners both early and late into the shift - with proper authorization, of course. The console is locked by default, and requires Silicon or RD-level access in order to manipulate the controls. 

Once access is obtained, Toxins Scientists* will be able to take properly-assembled TTV's and load them in. The L.A.M. can then be instructed to target one of the GPS coordinates visible on Lavaland.
##### * - anyone that obtains access to the machine's location and has a functional TTV can do this, too.
image

As is in the name, the L.A.M. delivers volatile ordnance at the request of the Mining team. In order to facilitate safe practices and employee retention, all variable adjustments are announced directly through the science department radio.  The console also requires a delay between starting the Firing process and its deployment. (Minimum 15 seconds)

image

A demonstration of Firing sequence w/ last-second Abort can be seen here:

https://www.youtube.com/watch?v=j2RWRjtX0yc

The L.A.M. has been installed inside the following stations' Toxins Testing Chambers, with minimal alterations:
- Eclipse
- IceBox
- Box
- Delta
- Meta

As part of the roll-out, the L.A.M. has also been setup at:
- Runtime
- CentComm

### Related changes made in this PR:
- If statement added to obj/item/gps Initialization, along with editing var data of Mining GPSes on Lavaland. This was a necessary change, as miners should not have to worry about naughty Scientists accidentally'ing their base because a GPS was left on. The If statement prevents the Lavaland GPSes from "appearing" to be on but actually being off, due to the way icon update is implemented.

## Why it's good for the game

This is a feature expansion that brings Science & Mining together in ultimately a net-positive direction. Toxins gains a new avenue to produce and use their bombs, without the massive hurdles currently plaguing their job.

### Ideas not attached to this PR, but should be under consideration:
- Adjusting Lavaland Bombcap Multiplier (since recent Maxcap nerf + Multiplier nerf 2 years ago to reduce lag)
- Bombing places other than Lavaland (and the station)
- A new machine that mimics L.A.M., but allows Cargo to drop supplies to their miners
- Allowing Emag'ing capabilities to the L.A.M. that doesn't undermine its existence

#### Changelog

:cl:  
rscadd: Adds the L.A.M. to Toxins.
rscdel: A table was removed to make way for one of the L.A.M.'s
tweak: Lavaland's Mining Base GPSes now start Off by default, for safety reasons.
imageadd: L.A.M. sprites created by Tipy
/:cl:

### CODE REVIEW

**Comment by Readystorm546** [comment]
I kid you not. This would get me back into toxins all by its self. This is an addition to toxins I never knew I wanted until now and I love it so much.

**Comment by maxion12345** [comment]
I feel like it would both be faster and safer to clear lavaland with plasmacutters, but it is also a unique and interesting idea that I don't see the harm in. Unless you get griffed by sir totsalot bombing you while mining. Maybe make it announce the GPS tag as well so you have time to yeet it and run, emag effect of glitching the tag/Coords or a silent launch

**Review by patpol4 — CHANGES_REQUESTED** [review]
I Like it.
Pubby doesn't exist anymore and meta had some updates, update your branch

**Comment by MonsWTF** [comment]
>Maybe make it announce the GPS tag as well so you have time to yeet it and run, emag effect of glitching the tag/Coords or a silent launch

image

Not going to toss in any Emag effects on the machine at this current time, but the GPS tag could be expanded. Also, when you first lock onto a GPS signal on Lavaland, the machine does shout the name over radio, so the miners hopefully shouldn't miss it.

**Comment by MonsWTF** [comment]
While it's not in this PR itself, I'd like to mention the topic I brought up of the Lavaland Bombcap Multiplier, which allows bombs detonated on the Lavaland Z-level a larger explosion radius than the default maxcap.

It was reduced from 3x to 2x about 2 years ago to reduce lag https://github.com/yogstation13/Yogstation/commit/c4fd8715a5f5e9c3c0446072a5d526ee3e82c8f6

Maxcap is also down from 5/10/20, to 3/6/12, all-in-all bringing Lavaland bombs from 15/30/60 to 6/12/24.

I'd propose a PR after this one to raise the Lavaland multiplier to 4x, so that it's at least back up to 12/24/48.

**Comment by Jarod1200** [comment]
good nerf to lavaland blood cult too

**Comment by DiamondSentinel** [comment]
Not that much of a nerf to lavaland cult as you have to target a GPS signal to bomb, just like BSA. 

Still, a good addition. Means that you don't have scientists giving security a heart attack by sprinting through the halls carrying a primed TTV. 

Does this machine have parts and can it be de/reconstructed?

**Comment by MonsWTF** [comment]
> Does this machine have parts and can it be de/reconstructed?

For balance reasons, not in this PR. I intentionally left that out to avoid the "optimized" situations of putting it somewhere more efficient, or somewhere to hide it that it can still keep sending bombs to annoy Lavaland miners, annoying Scientists with the radio chatter, etc.

**Review by patpol4 — APPROVED** [review]
Maps appear fine

File: code/game/machinery/sci_bombardment.dm [hunk 1]
```diff
@@ -0,0 +1,257 @@
...
+					locked = !locked
+					radio.talk_into(src, "Controls [locked ? "locked" : "unlocked"] by [I.registered_name].",)
+				else
+					to_chat(usr, "<span class='warning'>Access denied. Please seek assistance from station AI or Research Director.</span>")
+			update_icon()
+			. = TRUE
+		if("count")//Prompts user to change countdown timer (Minimum based on var/mincount)
+			if(locked)
+				return
+			var/a = text2num(stripped_input(usr, "Set a new countdown timer. (Minimum [mincount])", name, mincount))
```
Comment (ghost) [inline code/game/machinery/sci_bombardment.dm:211]: slap a max arg on this. Some enterprising shitters will eventually find the list of href exploits and force this to hit negative and instantly send the bomb somehow.

File: code/game/machinery/sci_bombardment.dm [hunk 2]
```diff
@@ -0,0 +1,257 @@
...
+		data["tcoords"] = tcoords
+		data["targetdest"] = targetdest
+
+	var/list/signals = list()
+	data["signals"] = list()
+
+	for(var/gps in GLOB.GPS_list)
+		var/obj/item/gps/G = gps
+		var/turf/pos = get_turf_global(G) // yogs - get_turf_global instead of get_turf
+		if(G.emped || !G.tracking || pos.z != 5)
```
Comment (ghost) [inline code/game/machinery/sci_bombardment.dm]: use the define for the z level and not a hard constant for maintainability please

**Review by ghost — CHANGES_REQUESTED** [review]

**Comment by MonsWTF** [comment]
Going to need to close to fix the conflicts and re-submit on my not-Master branch. Will eventually open new PR and refer to this one.

**Comment by Marmio64** [comment]
your video is a disappointment, wasn't even a last second abort, there was like, at least 3 seconds left before it launched.

**Comment by MonsWTF** [comment]
I believe it's all fixed now.

**Review by Partheo — APPROVED** [review]

**Review by Partheo — APPROVED** [review]

**Review by Amelia0010 — APPROVED** [review]
Ping me when the merge conflicts are gone and I'll merge this

**Comment by patpol4** [comment]
@MonsWTF just fix the conflicts and we will merge this

**Review by JamieD1 — APPROVED** [review]
Fix cornflakes

**Comment by patpol4** [comment]
Fixed the map conflicts because I didn't want to see this go to waste, also may have accidentally fucked something up with the js tgui file when merging so make sure to check if it's not wrong. @yogstation13/maintainers

**Pull request merged**